br: add a document for backup restore systables (#5797)
pingcap/docs: br/backup-and-restore-tool.md

| Use BR v4.0 to back up TiDB v4.0 | ✅ | ✅ | ✅ (If TiKV >= v4.0.0-rc.1, and if BR contains the [#233](https://github.com/pingcap/br/pull/233) bug fix and TiKV does not contain the [#7241](https://github.com/tikv/tikv/pull/7241) bug fix, BR will cause the TiKV node to restart.) |
| Use BR nightly or v5.0 to back up TiDB v4.0 | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) | ❌ (If the TiDB version is earlier than v4.0.9, the [#609](https://github.com/pingcap/br/issues/609) issue might occur.) |

### Backup and restore of system schemas

Before v5.1.0, BR filtered out data from the system schemas during the backup.

> Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the system tables are **not** restored by default during the **restore**. Only when the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter) is set will the system tables be restored to the temporary schemas and then restored to the system schemas by renaming the temporary schemas.

Reviewer: ```suggestion
Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the system tables are **not** restored by default during the **restore**. If you want the system tables to be restored to the system schemas, you need to set the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter). Then, the system tables are first restored to the temporary schemas and then to the system schemas (by renaming the temporary schemas).
```

> Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the system tables are **not** restored by default during the **restore**. If you want the tables to be restored to the system schemas, you need to set the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter). Then, the system tables are first restored to the temporary schemas and then to the system schemas (by renaming the temporary schemas).

Reviewer 2: ```suggestion
Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the tables in system schema are **not** restored by default during the **restore**. If you want the tables to be restored to the system schemas, you need to set the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter). Then, the system tables are first restored to the temporary schemas and then to the system schemas (by renaming the temporary schemas).
```

> Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the system tables are **not** restored by default during the **restore**. If you want the system tables to be restored to the system schemas, you need to set the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter). Then, the system tables are first restored to the temporary schemas and then to the system schemas (by renaming the temporary schemas).

Reviewer 2: ```suggestion
Since v5.1.0, BR **backups** all data by default, including the system schema (`mysql.*`). But to be compatible with the earlier versions of BR, the system tables are **not** restored by default during the **restore**. If you want the tables to be restored to the system schemas, you need to set the [`filter` parameter](/br/use-br-command-line-tool.md#back-up-with-table-filter). Then, the system tables are first restored to the temporary schemas and then to the system schemas (by renaming the temporary schemas).
```

Reviewer 2: some user might put the customer table in system schema.

> Meanwhile, special treatment is performed on the following system tables:

Reviewer: ```suggestion
In addition, TiDB performs special operations on the following system tables:
```

> - Tables related to statistical information: No restore is performed because the table ID of the statistical information has changed.

Reviewer: ```suggestion
- Tables related to statistical information are not restored, because the table ID of the statistical information has changed.
```

> - `tidb` and `global_variables` tables under the `mysql` schema: No restore is performed because the tables cannot be overwritten. For example, the overwriting of GC safepoint affects the cluster.

Reviewer: @3pointer PTAL whether this comment is correct or not. Thanks~
```suggestion
- `tidb` and `global_variables` tables in the `mysql` schema are not restored, because these tables cannot be overwritten. For example, overwriting these tables by the GC safepoint will affect the cluster.
```

> - The `user` table under the `mysql` schema: After restoring the `user` table, you need to manually execute the `FLUSH PRIVILEGE` command to make it take effect.

Reviewer: ```suggestion
- The restore of the `user` table in the `mysql` schema does not take effect until you manually execute the `FLUSH PRIVILEGE` command.
```
